python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -378,9 +511,18 @@ def prepareDcrInputs(self, templateCoadd, warpRefList, weightList):
         angleDict = {}
         psfSizeDict = {}
         for visitNum, warpExpRef in enumerate(warpRefList):
-            visitInfo = warpExpRef.get(tempExpName + "_visitInfo")
-            visit = warpExpRef.dataId["visit"]
-            psf = warpExpRef.get(tempExpName).getPsf()
+            if isinstance(warpExpRef, DeferredDatasetHandle):
+                # Gen 3 API
+                bbox = geom.Box2I(templateCoadd.getBBox().getMin(), geom.Extent2I(1, 1))
```

Comment:
watch DM-22363, which I hope to merge today. It will allow you to use a deferredDatasetHandle to get a component of a dataset directly. i.e. warpExpRef.get(component="visitInfo").